main.go

```diff
@@ -112,11 +112,15 @@ func runServer(c *cli.Context) {
 		log.Fatal(err)
 	}
 	err = rp.Initialize(reverseproxy.ReverseProxyConfig{
-		Router:          &r,
-		RequestIDHeader: c.String("request-id-header"),
-		FlushInterval:   time.Duration(c.Int("flush-interval")) * time.Millisecond,
-		DialTimeout:     time.Duration(c.Int("dial-timeout")) * time.Second,
-		RequestTimeout:  time.Duration(c.Int("request-timeout")) * time.Second,
+		Router:            &r,
+		RequestIDHeader:   c.String("request-id-header"),
+		FlushInterval:     time.Duration(c.Int("flush-interval")) * time.Millisecond,
+		DialTimeout:       time.Duration(c.Int("dial-timeout")) * time.Second,
+		RequestTimeout:    time.Duration(c.Int("request-timeout")) * time.Second,
+		ReadTimeout:       c.Duration("client-read-timeout") * time.Second,
```

Comment:
You dont need to do `* time.Second` since the flag is already a duration, you need to pass it directly.